File: cpp/leetcode3.cpp
#include <iostream>
#include <string.h>
using namespace std;
class Solution {
public:
    bool bucket[256];
    int maxans = 0;
    int lengthOfLongestSubstring(string s) {
        clearbucket();
        if(s.length()==0)   return 0;
        for(int i=0;i<s.length();i++)
        {
            int ans = 0;
            for(int j=i;j<s.length();j++)
            {
                if(!bucket[s[j]])
                {
                    ans++;
                    bucket[s[j]]=true;
                    if(j==s.length()-1)
                    {
                        clearbucket();
                        if(ans>maxans)
                        {
                            maxans=ans;
                        }
                    }
                }
                else
                {
                    clearbucket();
                    if(ans>maxans)
                    {
                        maxans=ans;
                    }
                    break;
                }
            }
        }
        return maxans;
    }
    void clearbucket()
    {
        memset(bucket,false,sizeof(bucket));
    }
};
int main(void)
{
    Solution a = Solution();
    cout<<a.lengthOfLongestSubstring("au")<<endl;
}

File: lua/leetcode3.lua
local bucket = {}
local str = {}
function Clearbucket()
    for i = 0,255 do
    bucket[i]=false;
    end
end
Clearbucket();
local s = io.read();
if(string.len(s)==1) then
    print(1)
    return
end
local ans=0;
local maxans=0;
for i=1,#s do
    ans=0;
    for j =i,#s do
        if(not bucket[string.byte(s,j,j)]) then
            ans=ans+1;
            bucket[string.byte(s,j,j)]=true
            if(j==string.len(s)) then
                Clearbucket();
                if(ans>maxans) then
                    maxans=ans;
                end
            end
        else
            Clearbucket();
            if(ans>maxans)  then
                maxans=ans;
            end
            break;
        end
    end
end
print(maxans)